### PR TITLE
Fix hotspots

### PR DESCRIPTION
**Pull request by LanchevychMaxym**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by tetyanakutsko — APPROVED** [review]
Great job 👏 👍 👌

File: EPlast/EPlast.BLL/Interfaces/Events/IActionManager.cs [hunk 1]
```diff
@@ -64,15 +65,15 @@ public interface IActionManager
         /// <returns>Status code of the subscribing on event operation.</returns>
         /// <param name="id">The Id of event</param>
         /// <param name="user">ClaimsPrincipal of logged in user</param>
```
Comment (bodnarira) [inline EPlast/EPlast.BLL/Interfaces/Events/IActionManager.cs:67]: Change description

File: EPlast/EPlast.BLL/Interfaces/Events/IActionManager.cs [hunk 2]
```diff
@@ -34,15 +35,15 @@ public interface IActionManager
         /// <param name="eventTypeId">The Id of event type</param>
         /// <param name="categoryId">The Id of event category</param>
         /// <param name="user">ClaimsPrincipal of logged in user</param>
-        Task<List<GeneralEventDTO>> GetEventsAsync(int categoryId, int eventTypeId, ClaimsPrincipal user);
+        Task<List<GeneralEventDTO>> GetEventsAsync(int categoryId, int eventTypeId, User user);
```
Comment (voroniak) [inline EPlast/EPlast.BLL/Interfaces/Events/IActionManager.cs:38]: It's better to use IEnumerable

File: EPlast/EPlast.BLL/Interfaces/Events/IActionManager.cs [hunk 3]
```diff
@@ -121,6 +122,6 @@ public interface IActionManager
 
         Task CheckEventsStatusesAsync();
 
-        Task<List<GeneralEventDTO>> GetEventsByStatusAsync(int categoryId, int typeId, int status, ClaimsPrincipal user);
+        Task<List<GeneralEventDTO>> GetEventsByStatusAsync(int categoryId, int typeId, int status, User user);
```
Comment (voroniak) [inline EPlast/EPlast.BLL/Interfaces/Events/IActionManager.cs:125]: )

**Review by voroniak — CHANGES_REQUESTED** [review]

File: EPlast/EPlast.BLL/Interfaces/AnnualReport/IAnnualReportService.cs [hunk 4]
```diff
@@ -15,14 +16,14 @@ public interface IAnnualReportService
         /// <returns>Annual report model</returns>
         /// <exception cref="System.UnauthorizedAccessException">Thrown when user hasn't access to annual report</exception>
         /// <exception cref="System.NullReferenceException">Thrown when annual report doesn't exist</exception>
-        Task<AnnualReportDTO> GetByIdAsync(ClaimsPrincipal claimsPrincipal, int id);
+        Task<AnnualReportDTO> GetByIdAsync(User user, int id);
 
         /// <summary>
         /// Method to get all reports that the user has access to
         /// </summary>
         /// <param name="claimsPrincipal">Authorized user</param>
```
Comment (bodnarira) [inline EPlast/EPlast.BLL/Interfaces/AnnualReport/IAnnualReportService.cs:24]: Description for param

File: EPlast/EPlast.BLL/Interfaces/Auth/IAuthService.cs [hunk 5]
```diff
@@ -142,7 +142,7 @@ public interface IAuthService
         /// </summary>
         /// <param name="claimsPrincipal"></param>
```
Comment (bodnarira) [inline EPlast/EPlast.BLL/Interfaces/Auth/IAuthService.cs:143]: Description

File: EPlast/EPlast.BLL/Interfaces/Auth/IAuthService.cs [hunk 6]
```diff
@@ -163,7 +163,7 @@ public interface IAuthService
...
```
Comment (bodnarira) [inline EPlast/EPlast.BLL/Interfaces/Auth/IAuthService.cs:164]: The same

File: EPlast/EPlast.BLL/Interfaces/Club/IClubAnnualReportService.cs [hunk 7]
```diff
@@ -18,14 +19,14 @@ public interface IClubAnnualReportService
         /// <returns>Annual report model</returns>
         /// <exception cref="System.UnauthorizedAccessException">Thrown when user hasn't access to annual report</exception>
         /// <exception cref="System.NullReferenceException">Thrown when annual report doesn't exist</exception>
-        Task<ClubAnnualReportDTO> GetByIdAsync(ClaimsPrincipal claimsPrincipal, int id);
+        Task<ClubAnnualReportDTO> GetByIdAsync(User claimsPrincipal, int id);
 
         /// <summary>
         /// Method to get all club reports that the user has access to
         /// </summary>
         /// <param name="claimsPrincipal">Authorized user</param>
```
Comment (bodnarira) [inline EPlast/EPlast.BLL/Interfaces/Club/IClubAnnualReportService.cs:27]: ?

File: EPlast/EPlast.BLL/Interfaces/Events/IActionManager.cs [hunk 8]
```diff
@@ -34,15 +35,15 @@ public interface IActionManager
         /// <param name="eventTypeId">The Id of event type</param>
         /// <param name="categoryId">The Id of event category</param>
         /// <param name="user">ClaimsPrincipal of logged in user</param>
```
Comment (bodnarira) [inline EPlast/EPlast.BLL/Interfaces/Events/IActionManager.cs:37]: ?

**Review by bodnarira — APPROVED** [review]

**Review by MykhailoKlapchuk — APPROVED** [review]

File: EPlast/EPlast.BLL/Interfaces/Events/IActionManager.cs (hunk 1 above)
Comment (LanchevychMaxym) [reply EPlast/EPlast.BLL/Interfaces/Events/IActionManager.cs:67]: fixed

File: EPlast/EPlast.BLL/Interfaces/Events/IActionManager.cs (hunk 2 above)
Comment (LanchevychMaxym) [reply EPlast/EPlast.BLL/Interfaces/Events/IActionManager.cs:38]: fixed

File: EPlast/EPlast.BLL/Interfaces/Events/IActionManager.cs (hunk 3 above)
Comment (LanchevychMaxym) [reply EPlast/EPlast.BLL/Interfaces/Events/IActionManager.cs:125]: fixed

File: EPlast/EPlast.BLL/Interfaces/AnnualReport/IAnnualReportService.cs (hunk 4 above)
Comment (LanchevychMaxym) [reply EPlast/EPlast.BLL/Interfaces/AnnualReport/IAnnualReportService.cs:24]: fixed

File: EPlast/EPlast.BLL/Interfaces/Auth/IAuthService.cs (hunk 6 above)
Comment (LanchevychMaxym) [reply EPlast/EPlast.BLL/Interfaces/Auth/IAuthService.cs:164]: fixed

File: EPlast/EPlast.BLL/Interfaces/Auth/IAuthService.cs (hunk 5 above)
Comment (LanchevychMaxym) [reply EPlast/EPlast.BLL/Interfaces/Auth/IAuthService.cs:143]: fixed

File: EPlast/EPlast.BLL/Interfaces/Club/IClubAnnualReportService.cs (hunk 7 above)
Comment (LanchevychMaxym) [reply EPlast/EPlast.BLL/Interfaces/Club/IClubAnnualReportService.cs:27]: fixed

File: EPlast/EPlast.BLL/Interfaces/Events/IActionManager.cs (hunk 8 above)
Comment (LanchevychMaxym) [reply EPlast/EPlast.BLL/Interfaces/Events/IActionManager.cs:37]: fixed

**Review by voroniak — APPROVED** [review]
LGTM!

**Pull request merged**